### PR TITLE
Port over Fox's refactored syndicate item refund from TG.

### PR DESCRIPTION
**Pull request by Anasari**
This port over fox's general refund proc from tgstation/#19472 in a slightly changed manner. Instead of it working with attackby() proc (By attacking the uplink), it work by instead clicking on a button called "Refund item in active hand" in the purchase menu. It also removes the old borg teleporter's refund proc and replace it with this one.

As per the original PR it also allows you to specify a refund path in case what's being bought =/= what's being refunded, and an optional refund amount if you wish to give a refund penalty.

This allows you to theoretically refund item using an uplink implant instead of relying on attacking it.

🆑 
feature: You can now refund borg teleporter and unused holoparasite injector using a "Refund item in active hand" button in the purchase menu
tweak: Borg teleporter no longer refunded by attacking an uplink with it.
🆑

### CODE REVIEW

File: code/datums/uplink_item.dm [hunk 1]
```diff
@@ -97,6 +100,7 @@ var/list/uplink_items = list()
 		if(I)
 			if(ishuman(user))
 				var/mob/living/carbon/human/A = user
+				log_game("[key_name(user)] purchased [I.name]")
```
Comment (Birdtalon) [inline code/datums/uplink_item.dm]: You're also adding this in #9056 in the same place.

**Comment by Anasari** [comment]
Fixed. I messed up in an earlier version and might've copied pasted the wrong file.

**Comment by Birdtalon** [comment]
FYI
https://github.com/tgstation/tgstation/pull/38272/

File: code/game/objects/items/devices/uplinks.dm [hunk 2]
```diff
@@ -142,6 +145,30 @@ var/list/world_uplinks = list()
 
 	return 1
 
+/obj/item/uplink/proc/refund(mob/user as mob)
+	var/obj/item/I = user.get_active_hand()
+	if(I) // Make sure there's actually something in the hand before even bothering to check
+		for(var/item in subtypesof(/datum/uplink_item))
```
Comment (Crazylemon64) [inline code/game/objects/items/devices/uplinks.dm]: https://github.com/tgstation/tgstation/pull/38272

**Comment by Anasari** [comment]
Ready for review

**Pull request merged**